File: MedClientSide/src/app/data/models/department.ts
import {User} from './user';
import {Service} from './service';
import {DepartmentService} from './department-service';
import {Schedule} from './schedule';

export class Department {
  public id: number;
  public departmentName: string;
  public description: string;
  public addressId: number;
  public scheduleId: number;
  public schedule: Schedule;
  public isVisible: boolean;
  public doctors: User[];
  public departmentServices: DepartmentService[];
}

File: WebAPI/MedClinicalAPI/Data/Models/Department.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace MedClinicalAPI.Data.Models
{
    [Table("Departments")]
    public class Department
    {
        public int Id { get; set; }
        public string DepartmentName { get; set; }
        public string Description { get; set; }
        public int AddressId { get; set; }
        public int ScheduleId { get; set; }
        public string PhotoPath { get; set; }

        // nav props
        public Schedule Schedule { get; set; }

        public Address Address { get; set; }
        public virtual ICollection<DepartmentService> DepartmentServices { get; set; }
        public virtual ICollection<User> Doctors { get; set; }
    }
}